src/nodes.coffee

```diff
@@ -3551,6 +3553,33 @@ exports.Code = class Code extends Base
 
     seenSuper
 
+  propagateLhs: ->
+    for {name} in @params when name instanceof Arr or name instanceof Obj
+      name.propagateLhs yes
+
+  astType: -> 'FunctionExpression'
```

Comment:
There's an argument to be made that bound (fat-arrow) functions should be type `ArrowFunctionExpression` instead of `FunctionExpression` since we transpile them to ES6 arrow functions

Currently instead all (non-class-method) functions are type `FunctionExpression` and there's an additional `bound` AST property to distinguish fat-arrow from skinny-arrow functions

I can try and make them `ArrowFunctionExpression` and check how well the Prettier and ESLint plugins handle that

Reply:
In astexplorer.net `const fn = () => {};` translates to `ArrowFunctionExpression`, so I guess we should be doing that. It shouldn’t be that hard, since we just look at whether it’s bound or not?

Reply:
Yup exactly. Before making that change here I'd just like to see what breaks in the Prettier and ESLint plugins when introducing `ArrowFunctionExpression`

Reply:
Sure, updated to use `ArrowFunctionExpression` if it's bound. And removed the `bound` AST property since it would be redundant (I think we'll want to re-add it to distinguish bound vs non-bound class methods, but we can do that when we get there)

It's on me to update those plugins anyway, I can do that independently of how we handle this PR